actions/marketplace.js

```diff
@@ -22,6 +22,13 @@ export function fetchPlugins() {
 
             return {plugins};
         } catch (error) {
+            const plugins = await Client4.getMarketplacePlugins(filter, true);
```

Comment:
We should handle a second error from this method, since it might be a real error.

Reply:
Separately, it still “feels” weird to trigger the second call based purely on an error response. Since we can use this error to detect that the remote marketplace is disabled, might we instead explicitly send down that bit in the client config and make just a single API call?

Reply:
The server determines if the `EnableRemoteMarketplace` setting disable and does not return an error but just the local plugins in that case. This error is not to detect that the remote marketplace is disabled but indeed unavailable.

What we could do to distinguish from other error is have the server return a different status code - maybe 503? - and we can check the type of error here. If it is indeed an unavailable service error, then retry with `local_only`, everything else would return the error as usual. Thoughts?

Reply:
Ah, yes, thank you for clarifying -- perhaps we just return an error with a known `id` and check for that?

Reply:
Updated - I've included to check for the id and directly return any other errors.